bundles/org.openhab.binding.entsoe/src/main/java/org/openhab/binding/entsoe/internal/EntsoeHandler.java

```diff
@@ -0,0 +1,265 @@
+            return;
+        }
+
+        config = getConfigAs(EntsoeConfiguration.class);
+
+        ZonedDateTime startUtc = currentUtcTimeWholeHours()
+                .minusDays(needToFetchHistoricDays() ? config.historicDays : 1).withHour(22);
+        ZonedDateTime endUtc = currentUtcTimeWholeHours().plusDays(1).withHour(22);
+
+        boolean needsUpdate = lastDayAheadReceived.equals(ZonedDateTime.of(LocalDateTime.MIN, ZoneId.of("UTC")))
+                || responseMap.isEmpty() || needToFetchHistoricDays(true);
+
+        boolean hasNextDayValue = needsUpdate ? false
+                : responseMap.entrySet().stream()
+                        .anyMatch(x -> x.getKey().getDayOfYear() == currentUtcTime().plusDays(1).getDayOfYear());
+        boolean readyForNextDayValue = needsUpdate ? true
+                : currentUtcTime().isAfter(currentUtcTime().withHour(config.spotPricesAvailableUtcHour));
+
+        // Update whole time series
+        if (needsUpdate || (!hasNextDayValue && readyForNextDayValue)) {
+            logger.debug("Updating timeseries");
+            Request request = new Request(config.securityToken, config.area, startUtc, endUtc);
+            Client client = new Client();
+            boolean success = false;
+
+            try {
+                responseMap = client.doGetRequest(request, 30000);
+                TimeSeries baseTimeSeries = new TimeSeries(EntsoeBindingConstants.TIMESERIES_POLICY);
+                for (Map.Entry<ZonedDateTime, Double> entry : responseMap.entrySet()) {
+                    BigDecimal kwhPrice = BigDecimal.valueOf(entry.getValue()).divide(new BigDecimal(1000), 10,
+                            RoundingMode.HALF_UP);
+                    baseTimeSeries.add(entry.getKey().toInstant(), getPriceState(kwhPrice));
+                }
+                lastDayAheadReceived = currentUtcTime();
+                sendTimeSeries(EntsoeBindingConstants.CHANNEL_SPOT_PRICES, baseTimeSeries);
+                updateCurrentHourState(EntsoeBindingConstants.CHANNEL_SPOT_PRICES);
+                scheduler.schedule(this::triggerChannelSpotPricesReceived, 0, TimeUnit.SECONDS);
+                success = true;
+            } catch (EntsoeResponseException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            } catch (EntsoeConfigurationException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+            } finally {
+                schedule(success);
+            }
+        } else {
+            logger.debug("Updating current hour");
+            updateCurrentHourState(EntsoeBindingConstants.CHANNEL_SPOT_PRICES);
+            schedule(true);
+        }
+    }
+
+    private State getPriceState(BigDecimal kwhPrice) {
+        if (baseUnit == null || fromUnit == null) {
```

Comment:
I believe neither `baseUnit` or `fromUnit` is never null since they are set in lines 69-70 and never changed:
```java
private Unit<Currency> baseUnit = CurrencyUnits.BASE_CURRENCY;
private Unit<Currency> fromUnit = new CurrencyUnit(EntsoeBindingConstants.ENTSOE_CURRENCY, null);
```

Reply:
No, see:
https://github.com/openhab/openhab-core/blob/a5c488d8c82f75b5ade26385c23b501c723fe637/bundles/org.openhab.core/src/main/java/org/openhab/core/library/unit/CurrencyUnits.java#L41

If EUR is not available because of (missing) unit settings, this should be covered by catching the IAE below.